.circle/index.py

```diff
@@ -67,6 +80,32 @@ def build_index(path_glob, output_path):
     print('Processed %s packs.' % (counter))
     print('Index data written to "%s".' % (output_path))
 
+
+def get_available_versions_for_pack(pack_ref):
+    """
+    Retrieve all the available versions for a particular pack.
+
+    NOTE: This function uses Github API.
+    """
+    url = ('https://api.github.com/repos/%s/%s-%s/tags' %
+           (EXCHANGE_NAME, EXCHANGE_PREFIX, pack_ref))
+    resp = SESSION.get(url)
+
+    if resp.status_code != 200:
+        print('Got non 200 response: %s' % (resp.text))
+        return None
+
+    versions = []
```

Comment:
Why not a list comprehension here?

Reply:
`.get(..., None)` is explicit, but redundant, since `.get()` returns `None` by default if the key doesn't exist.

Also, there is a potential bug here.

```python
item.get('name', None)
```

so you are protecting against the case where an `item` doesn't have a `name` key, which is good. But, if `name` doesn't exist, `.get()` returns `None`, which then has `startswith()` run on it. This will result in (I think) an `AttributeError`:

```
>>> a = {
...     'key': 'value'
... }
>>> a.get('name', None).startswith('v')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'startswith'
```

This logic is much better represented in a list comprehension:

```python
return list(reversed(sorted([
    item['name'].replace('v', '')
    for item in resp.json()
    if item.get('name', '').startswith('v')
])))
```

That fixes the bug and reads better IMO.

Reply:
Just a style preference, I like to use list comprehensions for simple stuff, this would likely result in two or more line long list comprehension, where, imo, regular for loop is a better fit.